src/instantiate.jl

```diff
@@ -0,0 +1,104 @@
+"""
+    struct UninstantiatedOptimizer
+        optimizer
```

Comment:
I know I suggested this name, but it's misleading because it's a function that returns an optimizer, not an optimizer itself. What about `optimizer_constructor`?

Reply:
This comment still applies.